packages/coding-style/src/Rector/ClassMethod/AnnotateThrowablesRector.php

```diff
@@ -0,0 +1,174 @@
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwarePhpDocTagNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use RuntimeException;
+
+/**
+ * Adds "throws" DocBlock to methods.
+ */
+final class AnnotateThrowablesRector extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Throw_::class];
+    }
+
+    /**
+     * From this method documentation is generated.
+     */
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Adds @throws DocBlock comments to methods that thrwo \Throwables.', [
+                new CodeSample(
+                     // code before
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                    ,
+                    // code after
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     * @throws \RuntimeException
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param Throw_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        $this->annotateMethod($node);
+
+        return $node;
+    }
+
+    private function isThrowableAnnotated(Throw_ $node): bool
+    {
+        $method = $node->getAttribute(AttributeKey::METHOD_NODE);
+
+        if ($method === null) {
+            throw new RuntimeException('This should not happen and is probably a bug. Please report it.');
+        }
+
+        /** @var PhpDocInfo $phpDocInfo */
+        $phpDocInfo = $method->getAttribute(AttributeKey::PHP_DOC_INFO);
+        $throwTags = $phpDocInfo->getTagsByName('throws');
+        $FQN = $this->buildFQN($node);
+
+        if (empty($throwTags)) {
+            return false;
+        }
+
+        /** @var AttributeAwarePhpDocTagNode $throwTag */
+        foreach ($throwTags as $throwTag) {
+            $throwClassName = $throwTag->value->type->name;
+            if ($throwClassName === $FQN) {
+                return true;
+            }
+
+            if (
+                ! Strings::contains($throwClassName, '\\') &&
+                Strings::contains($FQN, $throwClassName) &&
+                $this->isThrowableImported($node)
+            ) {
+                return true;
+            }
```

Comment:
This is really not needed, as the `$this->getName()` method always returns FQN name

Stripping `\\` of docblock should help:

```php
$throwClassName = trim($throwClassName, '\\');
if ($throwClassName === $FQN) {
    return true;
}
```

Reply:
Using

```php
        /** @var AttributeAwarePhpDocTagNode $throwTag */
        foreach ($throwTags as $throwTag) {
            // $thrownClassName = $throwTag->value->type->name;
            $thrownClassName = $this->getName($throwTag);
```

Causes an error:

> TypeError: Argument 1 passed to Rector\Core\Rector\AbstractRector::getName() must implement interface PhpParser\Node, instance of Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwarePhpDocTagNode given

Reply:
That's doc block tag, it's something else than php-parser node